Userland/Libraries/LibGfx/ImageFormats/PNGWriter.cpp

```diff
@@ -127,6 +129,41 @@ ErrorOr<void> PNGWriter::add_IHDR_chunk(u32 width, u32 height, u8 bit_depth, PNG
     return {};
 }
 
+ErrorOr<void> PNGWriter::add_acTL_chunk(u32 num_frames, u32 num_plays = 0)
+{
+    PNGChunk png_chunk { "acTL"_string };
+    TRY(png_chunk.add_as_big_endian(num_frames));
+    TRY(png_chunk.add_as_big_endian(num_plays));
+    TRY(add_chunk(png_chunk));
+    return {};
+}
+
+ErrorOr<void> PNGWriter::add_fcTL_chunk(u32 sequence_number, u32 width, u32 height, u16 delay_num = 1, u16 delay_den = 30, u8 dispose_op = 0, u8 blend_op = 0, u32 x_offset = 0, u32 y_offset = 0)
+{
+    // TODO: Constraints on frame regions
+    PNGChunk png_chunk { "fcTL"_string };
+    TRY(png_chunk.add_as_big_endian(sequence_number));
+    TRY(png_chunk.add_as_big_endian(width));
+    TRY(png_chunk.add_as_big_endian(height));
+    TRY(png_chunk.add_as_big_endian(x_offset));
+    TRY(png_chunk.add_as_big_endian(y_offset));
+    TRY(png_chunk.add_as_big_endian(delay_num));
+    TRY(png_chunk.add_as_big_endian(delay_den));
+    TRY(png_chunk.add_u8(dispose_op));
+    /* dispose_op values
+    0           APNG_DISPOSE_OP_NONE
+    1           APNG_DISPOSE_OP_BACKGROUND
+    2           APNG_DISPOSE_OP_PREVIOUS
+    */
+    TRY(png_chunk.add_u8(blend_op));
+    /* blend_op values
+    value
+    0       APNG_BLEND_OP_SOURCE
+    1       APNG_BLEND_OP_OVER*/
```

Comment:
We generally use `//` comments for everything except the license headers, but that might not be a big deal.